### PR TITLE
Cmake clang sanitizer

### PR DESCRIPTION
**Pull request by Kokan**
This make it possible to configure with a simple option (`sanitizer`) in case of cmake to use the clang address (or any other sanitizer).

https://clang.llvm.org/docs/AddressSanitizer.html

It is connected to the #2178 issue, which has a connected branch with memory issue fixes. I do rebase the current master from time to time, and since the last time (end of December) I got a lot more test failing, so I though I share a simple way to deploy it for others (and make my life easier not to switch, rebase branch).

This is cmake only, as I won't touch autotools before every unit tests passed.

What you actually need to set this up is: clang, cmake an example can be found on the original branch: https://github.com/Kokan/syslog-ng/blob/address-sanatize/.travis.yml#L157

### CODE REVIEW

**Comment by kira-syslogng** [comment]
Build SUCCESS

File: CMakeLists.txt [hunk 1]
```diff
@@ -334,6 +334,17 @@ endif()
 
 add_compile_options(${IMPORTANT_WARNINGS} ${ACCEPTABLE_WARNINGS} ${EXTRA_WARNINGS})
 
+# Sanatizer configuration
+set(SANITIZER "OFF" CACHE STRING "Enable clang sanitizer")
+
+set_property(CACHE SANITIZER PROPERTY STRINGS OFF address undefined)
+
+if (NOT ${SANITIZER} STREQUAL "OFF")
```
Comment (furiel) [inline CMakeLists.txt]: OFF, NO, etc evaluate to false. Anything else including `address` and `undefined` evaluates to true. So it is enough to simply write `if (SANITIZER)`.

File: CMakeLists.txt [hunk 2]
```diff
@@ -334,6 +334,17 @@ endif()
...
+  SET(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -O1 -fsanitize=${SANITIZER} -fno-omit-frame-pointer")
```
Comment (furiel) [inline CMakeLists.txt]: Why did you use -O1? Couldn't we more accurate results with `-O0`?

File: CMakeLists.txt (hunk 1 above)
Comment (Kokan) [reply CMakeLists.txt]: Thanks for the suggestion, fixed.

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Pull request merged**